client/widgets/CArtifactHolder.h

```diff
@@ -41,13 +41,44 @@ class CWindowWithArtifacts : public CArtifactHolder
 	void artifactAssembled(const ArtifactLocation &artLoc) override;
 };
 
-/// Artifacts can be placed there. Gets shown at the hero window
-class CArtPlace: public LRClickableAreaWTextComp
+class CArtPlace : public LRClickableAreaWTextComp
 {
+protected:
 	CAnimImage *image;
+	virtual void createImage()=0;
+public:
+	const CArtifactInstance * ourArt; // should be changed only with setArtifact()
+
+	CArtPlace(Point position, const CArtifactInstance * Art = nullptr); //c-tor
+	void clickLeft(tribool down, bool previousState) override;
+	void clickRight(tribool down, bool previousState) override;
+
+	virtual void setArtifact(const CArtifactInstance *art)=0;
+};
+
+class CCommanderArtPlace : public CArtPlace
+{
+protected:
+	const CGHeroInstance * commanderOwner;
+	ArtifactPosition commanderSlotID;
```

Comment:
I decided to separate this from CHeroArtPlace::slotID, as the latter is managed magically by CArtifactsOfHero, and I did not want to set it up in CArtPlace constructor (which is now a base class)